core/internal/runhistoryreader/reader.go

```diff
@@ -226,19 +270,32 @@ func (h *HistoryReader) initParquetFiles(ctx context.Context) error {
 	for i, url := range signedUrls {
 		var parquetFile *pqarrow.FileReader
 
-		// When the user doesn't specify any keys,
-		// It is faster to download the entire parquet file
-		// and process it locally.
-		if len(h.keys) == 0 {
-			tmpDir := os.TempDir()
-			fileName := fmt.Sprintf("run_history_%d.parquet", i)
+		dir := os.Getenv("WANDB_CACHE_DIR")
+		if dir == "" {
+			dir, _ = os.UserCacheDir()
+			dir = filepath.Join(dir, "wandb", "runhistory")
+		}
+		if dir == "" {
+			return fmt.Errorf("failed to get runhistory cache directory")
+		}
```

Comment:
This looks like it would be cleaner as a helper function with early returns for error conditions.